cloud/shared/bin/doctor_impl

```diff
@@ -0,0 +1,221 @@
+#! /usr/bin/env python3
+
+import platform
+import os
+import subprocess
+
+detected_platform = platform.system()
+is_linux = detected_platform == 'Linux'
+is_mac = detected_platform == 'Darwin'
```

Comment:
do we need a wsl one?

Reply:
I'm preeeetty sure WSL will return 'Linux"